File: 0038-count-and-say/0038-count-and-say.js
/**
 * @param {number} n
 * @return {string}
 */
var countAndSay = function(n) {
    var s='1'
    for(let k =0; k<n-1;k++){
        var sCopy=[]
        var i=0
        var j=0
        while (j<s.length){
            if(s[i]==s[j]){
                j++
            }
            else{
                sCopy=`${sCopy}${j-i}`
                sCopy=`${sCopy}${s[i]}`
                i=j
            }
        }
        sCopy=`${sCopy}${j-i}`
        sCopy=`${sCopy}${s[i]}`
        s=sCopy
        
    }
    return s
    
};

File: 0038-count-and-say/0038-count-and-say.py
'''
   1  n=1
   11 n=2
   2 1   n=3
   1 2 1 1     n=4
   111221      n=5
   312211 n=6
'''
class Solution:
    def countAndSay(self, n: int) -> str:
        
        s='1'
        for _ in range(n-1):
            sCopy=""
            i=0
            j=0
            while j<len(s):
                if s[i]==s[j]:
                    j+=1
                else:
                    sCopy+=str(j-i)
                    sCopy+=s[i]
                    i=j
            if i!=j:
                sCopy+=str(j-i)
                sCopy+=s[i]  
            s=sCopy
        return s
        